tests/Doctrine/Tests/KeyValueStore/Storage/AmazonDynamoDbTest.php

```diff
@@ -0,0 +1,278 @@
+<?php
+
+namespace Doctrine\Tests\KeyValueStore\Storage;
+
+use Doctrine\KeyValueStore\Storage\AmazonDynamoDbStorage;
+
+class AmazonDynamoDbTest extends \PHPUnit_Framework_TestCase
+{
+    private function getDynamoDbMock($methods = [])
+    {
+        $client = $this->getMockBuilder('Aws\DynamoDb\DynamoDbClient')->disableOriginalConstructor();
+
+        if (count($methods)) {
+            $client->setMethods($methods);
+        }
+
+        return $client->getMock();
+    }
+
+    private function getDynamoDbResultMock($methods = [])
+    {
+        $result = $this->getMockBuilder('Aws\Result')->disableOriginalConstructor();
+
+        if (count($methods)) {
+            $result->setMethods($methods);
+        }
+
+        return $result->getMock();
+    }
+
+    public function testTheStorageName()
+    {
+        $client = $this->getDynamoDbMock();
+
+        $storage = new AmazonDynamoDbStorage($client);
+        $this->assertSame('amazon_dynamodb', $storage->getName());
+    }
+
+    public function testDefaultKeyName()
+    {
+        $client = $this->getDynamoDbMock();
+        $storage = new AmazonDynamoDbStorage($client);
+        $this->assertAttributeSame('Id', 'defaultKeyName', $storage);
+    }
+
+    public function testThatTableKeysInitiallyEmpty()
+    {
+        $client = $this->getDynamoDbMock();
+        $storage = new AmazonDynamoDbStorage($client);
+        $this->assertAttributeSame([], 'tableKeys', $storage);
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage The key must be a string, got "array" instead.
+     */
+    public function testDefaultKeyCannotBeSomethingOtherThanString()
+    {
+        $client = $this->getDynamoDbMock();
+        new AmazonDynamoDbStorage($client, null, []);
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
```

Comment:
Please use `$this->setExpectedException()` instead, so you know exactly at which point in the execution an exception should be thrown